File: backend/internal/models/non_spotify.go
package models

import (
	"time"
)

type NonSpotifyUser struct {
	ID         string    `gorm:"primaryKey" json:"id"`
	Passphrase string    `gorm:"not null" json:"-"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

type NonSpotifyPlaylist struct {
	ID          string    `gorm:"primaryKey" json:"id"`
	UserID      string    `gorm:"not null" json:"user_id"`
	Name        string    `gorm:"not null" json:"name"`
	Genre       string    `json:"genre"`
	Description string    `json:"description"`
	ImageURL    string    `json:"image_url"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type NonSpotifyPlaylistTrack struct {
	ID              string    `gorm:"primaryKey" json:"id"`
	PlaylistID      string    `gorm:"not null" json:"playlist_id"`
	Title           string    `gorm:"not null" json:"title"`
	Artist          string    `gorm:"not null" json:"artist"`
	AddedToPlaylist bool      `gorm:"default:false" json:"added_to_playlist"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

type NonSpotifyPlaylistSeedTrack struct {
	ID         string    `gorm:"primaryKey" json:"id"`
	PlaylistID string    `gorm:"not null" json:"playlist_id"`
	Title      string    `gorm:"not null" json:"title"`
	Artist     string    `gorm:"not null" json:"artist"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

type NonSpotifyPlaylistWithTracks struct {
	NonSpotifyPlaylist
	Tracks     []NonSpotifyPlaylistTrack     `json:"tracks"`
	SeedTracks []NonSpotifyPlaylistSeedTrack `json:"seed_tracks"`
}

File: frontend/src/types/non-spotify.ts
export type NonSpotifyUser = {
    id: string;
    created_at: string;
    updated_at: string;
};

export type NonSpotifyPlaylist = {
    id: string;
    user_id: string;
    name: string;
    genre: string;
    description: string;
    image_url: string;
    created_at: string;
    updated_at: string;
};

export type NonSpotifyPlaylistTrack = {
    id: string;
    playlist_id: string;
    title: string;
    artist: string;
    added_to_playlist: boolean;
    created_at: string;
    updated_at: string;
};

export type NonSpotifyPlaylistSeedTrack = {
    id: string;
    playlist_id: string;
    title: string;
    artist: string;
    created_at: string;
    updated_at: string;
};

export type NonSpotifyPlaylistWithTracks = NonSpotifyPlaylist & {
    tracks: NonSpotifyPlaylistTrack[];
    seed_tracks: NonSpotifyPlaylistSeedTrack[];
};

export type RegisterRequest = {
    user_id: string;
};

export type RegisterResponse = {
    user_id: string;
    passphrase: string;
};

export type VerifyRequest = {
    user_id: string;
    passphrase: string;
};

export type SeedTrack = {
    title: string;
    artist: string;
};

export type GeneratePlaylistRequest = {
    seed_tracks: SeedTrack[];
    genre: string;
};

export type UpdateTrackStatusRequest = {
    added_to_playlist: boolean;
};